src/curl.c

```diff
@@ -363,6 +364,62 @@ int swupd_curl_get_file(const char *url, char *filename, struct file *file,
 	return err;
 }
 
+void swupd_curl_test_resume(void)
+{
+#define RESUME_BYTE_RANGE 2
+
+	CURLcode curl_ret;
+	char *version_string = NULL;
+	char *url = NULL;
+
+	if (!curl) {
+		abort();
+	}
+
+	curl_easy_reset(curl);
+	version_string = malloc(LINE_MAX);
+	if (version_string == NULL) {
+		abort();
+	}
+
+	string_or_die(&url, "%s/version/format%s/latest", version_url, format_string);
+
+	curl_ret = curl_easy_setopt(curl, CURLOPT_RESUME_FROM_LARGE, (curl_off_t)RESUME_BYTE_RANGE);
+	if (curl_ret != CURLE_OK) {
+		goto exit;
+	}
+
+	curl_ret = curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, swupd_download_version_to_memory);
+	if (curl_ret != CURLE_OK) {
+		goto exit;
+	}
+	curl_ret = curl_easy_setopt(curl, CURLOPT_WRITEDATA, (void *)version_string);
```

Comment:
Ah good findings there, I'd agree the current solution is simpler and more effective since we can't make any definitive decision based on the header anyway.